### PR TITLE
Dev 386: Spurious errors about labels in shadow DOM #386

### PR DESCRIPTION
**Pull request by shunguoy**

### CODE REVIEW

File: accessibility-checker-engine/src/v2/checker/accessibility/rules/rpt-input-rules.ts [hunk 1]
```diff
@@ -78,7 +78,8 @@ let a11yRulesInput: Rule[] = [
             if (textTypes.indexOf(type) !== -1) { // If type is in the list
                 // Get only the non-hidden labels for element, in the case that an label is hidden then it is a violation
                 // Note: label[for] does not work for ARIA-defined inputs
-                let labelElem = ruleContext.hasAttribute("role") ? null : RPTUtil.getLabelForElementHidden(ruleContext, true);
+                //let labelElem = ruleContext.hasAttribute("role") ? null : RPTUtil.getLabelForElementHidden(ruleContext, true);
+                let labelElem = RPTUtil.getLabelForElementHidden(ruleContext, true);
```
Comment (tombrunet) [inline accessibility-checker-engine/src/v2/checker/accessibility/rules/rpt-input-rules.ts:82]: label/for is only valid for HTML native elements. When `role` is specified, this form of labelling is not supported - they must use the `aria-label` / `aria-labelledby` pattern. Original code was correct.

File: accessibility-checker-engine/src/v2/checker/accessibility/util/legacy.ts [hunk 2]
```diff
@@ -1616,22 +1616,21 @@ export class RPTUtil {
      * @memberOf RPTUtil
      */
     public static getLabelForElementHidden(element: Element, ignoreHidden) {
-
         // Check if the global RPTUtil_LABELS hash is available, as this will contain the label nodes based on
         // for attribute.
-        if (!RPTUtil.getCache(element.ownerDocument,"RPTUtil_LABELS", null)) {
+        //if (!RPTUtil.getCache(element.ownerDocument,"RPTUtil_LABELS", null)) {
+        let root = element.getRootNode();
+        if (!RPTUtil.setCache((root.nodeType === 11)? <ShadowRoot>root : <Document>root, "RPTUtil_LABELS", null)) {
```
Comment (tombrunet) [inline accessibility-checker-engine/src/v2/checker/accessibility/util/legacy.ts:1623]: I think you meant `getCache` here instead of `setCache`

**Review by tombrunet — CHANGES_REQUESTED** [review]

File: accessibility-checker-engine/src/v2/checker/accessibility/rules/rpt-input-rules.ts (hunk 1 above)
Comment (shunguoy) [reply accessibility-checker-engine/src/v2/checker/accessibility/rules/rpt-input-rules.ts:82]: didn't realize this.  restored.

File: accessibility-checker-engine/src/v2/checker/accessibility/util/legacy.ts (hunk 2 above)
Comment (shunguoy) [reply accessibility-checker-engine/src/v2/checker/accessibility/util/legacy.ts:1623]: cleanup error. fixed.

**Review by tombrunet — APPROVED** [review]

**Pull request merged**